### PR TITLE
cli: fix builder persistent flag

### PR DESCRIPTION
**Pull request by crazy-max**
`--builder` flag should not be persistent for all subcommands.

Before

```console
$ docker buildx version --help

Usage:  docker buildx version

Show buildx version information

Options:
      --builder string   Override the configured builder instance
```

Now

```console
$ docker buildx version --help

Usage:  docker buildx version

Show buildx version information
```

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

**Comment by tonistiigi** [comment]
It is somewhat nice though that you can do `buildx="buildx --builder=foo"` and then `$buildx build` or any other command.

**Comment by crazy-max** [comment]
will instead hide the builder flags for subcommands that don't need it as discussed.

**Comment by crazy-max** [comment]
well, looks like we can't hide a persistent flag (yet):
* https://github.com/spf13/cobra/issues/1132

File: go.mod [hunk 1]
```diff
@@ -57,6 +57,7 @@ require (
 
 replace (
 	github.com/docker/cli => github.com/docker/cli v20.10.3-0.20210702143511-f782d1355eff+incompatible
+	github.com/docker/cli-docs-tool => github.com/crazy-max/docker-cli-docs-tool v0.0.0-20211103173750-b8d48770fed8
```
Comment (crazy-max) [inline go.mod:60]: needs upstream pr https://github.com/docker/cli-docs-tool/pull/10

File: util/cobrautil/cobrautil.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import "github.com/spf13/cobra"
+
+// HideInheritedFlags hides inherited flags
+func HideInheritedFlags(cmd *cobra.Command, hidden ...string) {
+	for _, h := range hidden {
+		// we could use cmd.SetHelpFunc to override the helper
+		// but, it's not enough because we also want the generated
+		// docs to be updated, so we override the flag instead
+		cmd.Flags().String(h, "", "")
+		_ = cmd.Flags().MarkHidden(h)
```
Comment (crazy-max) [inline util/cobrautil/cobrautil.go:12]: would have prefer to use `SetHelpFunc` like:

```go
cmd.SetHelpFunc(func(ccmd *cobra.Command, args []string) {
	for _, h := range hidden {
		_ = ccmd.Flags().MarkHidden(h)
	}
	cmd.SetHelpFunc(nil)
	cmd.HelpFunc()(ccmd, args)
})
```

but it cannot be handled for docs generation unfortunately.

File: commands/imagetools/inspect.go [hunk 3]
```diff
@@ -59,10 +60,10 @@ func inspectCmd(dockerCli command.Cli) *cobra.Command {
 	}
 
 	flags := cmd.Flags()
-
 	flags.BoolVar(&options.raw, "raw", false, "Show original JSON manifest")
 
-	_ = flags
+	// hide builder persistent flag for this command
+	cobrautil.HideInheritedFlags(cmd, "builder")
```
Comment (tonistiigi) [inline commands/imagetools/inspect.go:66]: I found an use-case for this in #825

Comment (crazy-max) [reply commands/imagetools/inspect.go]: rebased since latest changes

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**